In [1]:
# Import Data Manipulation Library
import numpy as np
import pandas as pd
# Import Data Visualization Library
import seaborn as sns
import matplotlib.pyplot as plt
# Import Filter Warning Libraries
# Filterwarning :---> This Will Hide Alll Warning Messages 
import warnings
warnings.filterwarnings('ignore')
# Import Data Logging
import logging
logging.basicConfig(level=logging.INFO,
                    filemode = 'w',
                    filename= 'app.log',
                    format= '%(asctime)s - %(levelname)s - %(message)s')

In [2]:
logging.info('Please Upload Dataset for EDA...')

In [3]:
data = pd.read_csv('bank-full.csv',sep = ';')

In [4]:
logging.info('The Dataset is Successfully Uploaded')

In [5]:
# Import Data Sample
data.sample(frac = 1)

age          job   marital  education default  balance housing loan  \
25      44       admin.   married  secondary      no     -372     yes   no   
32746   43  blue-collar  divorced  secondary      no     -447     yes   no   
37269   21  blue-collar    single  secondary      no      292     yes  yes   
5298    31   management   married   tertiary      no     5889     yes   no   
30032   53  blue-collar   married    primary      no     2135      no   no   
...    ...          ...       ...        ...     ...      ...     ...  ...   
7002    49  blue-collar   married  secondary      no      457     yes  yes   
27393   32  blue-collar    single  secondary      no     1047      no   no   
21145   47     services   married  secondary      no     -150      no   no   
19696   50  blue-collar   married  secondary      no      283     yes   no   
45103   33       admin.   married  secondary      no       51     yes  yes   

         contact  day month  duration  campaign  pdays  previous poutcome    y  
25       unknown    5   may       172         1     -1         0  unknown   no  
32746   cellular   17   apr        85         1     -1         0  unknown   no  
37269  telephone   13   may       148         4     -1         0  unknown   no  
5298     unknown   23   may       237         1     -1         0  unknown   no  
30032   cellular    4   feb       152         2     -1         0  unknown   no  
...          ...  ...   ...       ...       ...    ...       ...      ...  ...  
7002     unknown   28   may       189         2     -1         0  unknown   no  
27393   cellular   21   nov        89         2     -1         0  unknown   no  
21145   cellular   14   aug        52         4     -1         0  unknown   no  
19696   cellular    7   aug       226         3     -1         0  unknown   no  
45103   cellular   25   oct       253         1    284         3  success  yes  

[45211 rows x 17 columns]

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45211 entries, 0 to 45210
Data columns (total 17 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   age        45211 non-null  int64 
 1   job        45211 non-null  object
 2   marital    45211 non-null  object
 3   education  45211 non-null  object
 4   default    45211 non-null  object
 5   balance    45211 non-null  int64 
 6   housing    45211 non-null  object
 7   loan       45211 non-null  object
 8   contact    45211 non-null  object
 9   day        45211 non-null  int64 
 10  month      45211 non-null  object
 11  duration   45211 non-null  int64 
 12  campaign   45211 non-null  int64 
 13  pdays      45211 non-null  int64 
 14  previous   45211 non-null  int64 
 15  poutcome   45211 non-null  object
 16  y          45211 non-null  object
dtypes: int64(7), object(10)
memory usage: 5.9+ MB


## Data Insights:

1. The Dataset Contains 45211 Rows and 17 Columns

2. Seven Numerical Columns and 10 Categorical Columns We have, Outof Which y is the target Column.